app/scripts/modules/kubernetes/src/loadBalancer/details/details.controller.ts

```diff
@@ -70,35 +57,33 @@ class KubernetesLoadBalancerDetailsController implements IController {
   public editLoadBalancer(): void {
     KubernetesManifestCommandBuilder.buildNewManifestCommand(
       this.app,
-      this.loadBalancer.manifest,
+      this.manifest.manifest,
       this.loadBalancer.moniker,
       this.loadBalancer.account,
     ).then(builtCommand => {
       ManifestWizard.show({ title: 'Edit Manifest', application: this.app, command: builtCommand });
     });
   }
 
-  private extractLoadBalancer(): void {
+  private extractLoadBalancer({ accountId, name, region }: ILoadBalancerFromStateParams): void {
     const rawLoadBalancer = this.app.getDataSource('loadBalancers').data.find((test: ILoadBalancer) => {
-      return (
-        test.name === this.loadBalancerFromParams.name &&
-        test.account === this.loadBalancerFromParams.accountId &&
-        test.region === this.loadBalancerFromParams.region
-      );
+      return test.name === name && test.account === accountId && test.region === region;
     });
 
-    if (rawLoadBalancer) {
-      this.state.loading = false;
-      this.loadBalancer = copy(rawLoadBalancer) as IKubernetesLoadBalancer;
-      this.loadBalancer.namespace = rawLoadBalancer.region;
-      this.loadBalancer.displayName = rawLoadBalancer.manifest.metadata.name;
-      this.loadBalancer.kind = rawLoadBalancer.manifest.kind;
-      this.loadBalancer.apiVersion = rawLoadBalancer.manifest.apiVersion;
-
-      this.app.getDataSource('loadBalancers').onRefresh(this.$scope, () => this.extractLoadBalancer());
-    } else {
-      this.autoClose();
+    if (!rawLoadBalancer) {
+      return this.autoClose();
     }
+
+    ManifestReader.getManifest(accountId, region, name).then((manifest: IManifest) => {
+      this.manifest = manifest;
+      this.loadBalancer = {
+        ...rawLoadBalancer,
+        apiVersion: manifest.manifest.apiVersion,
+        displayName: manifest.manifest.metadata.name,
+        namespace: rawLoadBalancer.region,
```

Comment:
@ezimanyi I didn't touch this because this diff was already getting out of control, but since we add these three fields to most of the infrastructure types, feels like we should either:

(1) Just read them directly off the manifest in the templates that need them
(2) Have Clouddriver include them in the payload

Originally, I would have said that going with (1) precludes us from future refactors in which the details sections that actually need the full manifest ("Edit" action button, for instance) could just fetch it for themselves on-demand, but since we need to respect the contract of the annotations-driven UI and so will need the manifest immediately to render any number of custom sections, I am now leaning toward (1).

Reply:
Assuming that all those templates will need the manifest anyway (which it seems they will based on the annotation-driven UI) I agree that going with (1) makes more sense.